llvm/test/CodeGen/RISCV/rvv/vsetvli-insert-crossbb.ll

```diff
@@ -328,7 +328,8 @@ define <vscale x 1 x double> @test8(i64 %avl, i8 zeroext %cond, <vscale x 1 x do
 ; CHECK-NEXT:    csrr a2, vlenb
 ; CHECK-NEXT:    slli a2, a2, 1
 ; CHECK-NEXT:    sub sp, sp, a2
-; CHECK-NEXT:    vsetvli s0, a0, e64, m1, ta, ma
+; CHECK-NEXT:    mv s0, a0
+; CHECK-NEXT:    vsetvli zero, a0, e64, m1, ta, ma
```

Comment:
The mv here comes from the call below which clobbers a0, but I don't think this is a regression anyway since we're removing the true dependency

Reply:
Agree. And I think the hardware may implement mv elimination.

Reply:
in-order cores can't really implement mv elimination since it requires register renaming.